### DISCUSSION TITLE
Why Select component can be empty when isRequired property exist?

### DISCUSSION THREAD

**Discussion by jkiss** [heroui-inc/heroui#2476] [Help] [answered]
I set `isRequired` prop on Select component, BUT when I click the same item, Select will be empty, I think this behavior is weird, right?

**Comment by alphaxek** [comment]
Hi @jkiss, can you please share your code snippet, are you passing the data to it?

**Reply by jkiss** [reply]
reproduce link:  https://codesandbox.io/p/devbox/jolly-rhodes-2742hf?file=%2Fapp%2Fpage.tsx%3A24%2C57

**Comment by wingkwong** [comment]
If possible, please share a video demo. When you click the same item again, it will unselect so the select will be empty.

**Reply by jkiss** [reply]
reproduce link:  https://codesandbox.io/p/devbox/jolly-rhodes-2742hf?file=%2Fapp%2Fpage.tsx%3A24%2C57

**Reply by wingkwong** [reply]
@jkiss The behaviour is correct. 

When you add `isRequired` to `Select`, it will have a danger asterisk at the end of the label and the select will be required. Since you don't have placeholder, you won't be able to see that `danger asterisk`. Here's an example how it looks like.

![image](https://github.com/nextui-org/nextui/assets/35857179/c03f9baf-3712-498c-969d-195d621c8ce4)

When you click the item in the first time, you select this item. When you click the same item again, it will unselect so the select will be empty. If you wrap it by `<form>`, when you submit an empty selection, you should be able to see `please fill in this field`.

![image](https://github.com/nextui-org/nextui/assets/35857179/eaf3acca-2a01-4f6d-ab5c-d9ede30bd1c5)

**Reply by jkiss** [reply]
ok...by the way, how can I disable the behavior that unselect item when click itself again? I just want there must be at least one value 😄

**Reply by wingkwong** [reply]
You can simply add a condition before calling `setInputLang`

**Reply by jkiss** [reply]
a little bit not easy to use, I also think click same item make it empty is NOT a normal behavior🤔, thanks for your response

**Comment by U29** [comment, marked as answer]
You can set the `disallowEmptySelection={true}` prop if you want to.
